routers/main.js

```diff
@@ -44,32 +44,45 @@ router.get('/createSuggestion',
 		req.session.title = req.query.title;
 		req.session.description = req.query.description;
 		req.session.state = req.query.state;
+		req.session.originalUrl = '/createSuggestion';
 		next();
 	},
 	passport.authenticate('github', {scope: [ 'repo' ]}));
 
 router.get('/github_callback', passport.authenticate('github', {failureRedirect: '/authKO'}), (req, res) => {
+
 	const suggestionsController = new SuggestionsController(config);
-	suggestionsController.createPullRequest(process.env.npm_config_ludwig_accessToken, req.session.title, req.session.description, req.session.state, res, config.github.branch);
+	if (req.session.originalUrl === '/createSuggestion') {
+		suggestionsController.createPullRequest(process.env.npm_config_ludwig_accessToken, req.session.title, req.session.description, req.session.state, res, config.github.branch);
```

Comment:
Why do you need to pass so many information while you instantiated the target object two lines above?!